paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -504,21 +541,26 @@ def downscale_aws_resource(self, filtered_slaves, current_capacity, target_capac
                         break
                 else:
                     break
-            try:
+
+            capacity_diff = new_capacity - current_capacity
+            self.log.info("Starting async kill for %s" % slave_to_kill.hostname)
+            # My understanding is that ensure_future will actually start running the coroutine
+            #  (gracefully_terminate_slave), until it hits something that sleeps, then the loop
+            #  can continue and we start killing the next slave
+            terminate_tasks[slave_to_kill.hostname] = asyncio.ensure_future(
                 self.gracefully_terminate_slave(
                     slave_to_kill=slave_to_kill,
-                    current_capacity=current_capacity,
-                    new_capacity=new_capacity,
-                )
-                killed_slaves += 1
-            except HTTPError:
-                # Something wrong draining host so try next host
-                continue
-            except FailSetResourceCapacity:
-                break
+                    capacity_diff=capacity_diff,
+                    timer=timer,
+                ),
+            )
+            killed_slaves += 1
 
             current_capacity = new_capacity
             mesos_state = get_mesos_master().state_summary()
+            # It's not obvious what this loop is doing: it is updating the task counts for
+            #  for the slaves in filtered_sorted_slaves, so next time through the loop
+            #  we can re-sort
```

Comment:
Just delete this. It looks like it is leftover from when it used to be the fitness stuff

Reply:
Yep as mentioned on #1560 this can be removed in this PR